lib/data_processor.rb

```diff
@@ -0,0 +1,66 @@
+require 'open-uri'
+require 'csv'
+
+class DataProcessor
+
+  def initialize(user_x, user_y, url)
+    @user_x = Float(user_x)
+    @user_y = Float(user_y)
+    @url = URI.parse(url)
+
+  rescue => ex
```

Comment:
Why rescue all types of exceptions here?

Reply:
Your are right, I will catch specific exceptions